org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinLateinitFilterTest.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2018 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Fabian Mastenbroek - implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.internal.analysis.filter;
+
+import org.jacoco.core.internal.instr.InstrSupport;
+import org.junit.Test;
+import org.objectweb.asm.Label;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.AbstractInsnNode;
+import org.objectweb.asm.tree.LabelNode;
+import org.objectweb.asm.tree.MethodNode;
+
+import static org.junit.Assert.*;
+
+public class KotlinLateinitFilterTest implements IFilterOutput {
+	private final KotlinLateinitFilter filter = new KotlinLateinitFilter();
+
+	private final MethodNode m = new MethodNode(InstrSupport.ASM_API_VERSION, 0,
+			"name", "()V", null, null);
+
+	private AbstractInsnNode fromInclusive;
+	private AbstractInsnNode toInclusive;
+
+	@Test
+	public void kotlinc() {
+		final Label l1 = new Label();
+		final Label l2 = new Label();
+
+		m.visitLabel(l1);
+		m.visitVarInsn(Opcodes.ALOAD, 0);
+		m.visitFieldInsn(Opcodes.GETFIELD, "com/better/alarm/background/VibrationService", "wakeLock", "Landroid/os/PowerManager$WakeLock;");
+		m.visitInsn(Opcodes.DUP);
+		m.visitJumpInsn(Opcodes.IFNONNULL, l2);
+		m.visitLdcInsn("wakelock");
+		m.visitMethodInsn(Opcodes.INVOKESTATIC, "kotlin/jvm/internal/Intrinsics",
+				"throwUninitializedPropertyAccessException", "Ljava/lang/String;", false);
+		m.visitLabel(l2);
+		m.visitMethodInsn(Opcodes.INVOKEVIRTUAL, "android/os/PowerManager$WakeLock", "acquire",
+				"", false);
+		filter.filter(m, new FilterContextMock(), this);
+
+		assertEquals(((LabelNode) l2.info).getPrevious().getPrevious().getPrevious(), fromInclusive);
```

Comment:
Sequence of `.getPrevious()` can be simplified by saving reference on instruction during construction as `expectedFrom = m.instructions.getLast()` - see for example https://github.com/jacoco/jacoco/blob/c8b8c9d3b7605fc09517c78f385fe31dccf04e31/org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/TryWithResourcesJavacFilterTest.java#L150